src/Gui/MDIView.cpp

```diff
@@ -448,4 +448,14 @@ void MDIView::setCurrentViewMode(ViewMode mode)
     }
 }
 
+QString MDIView::buildWindowTitle() {
+    QString windowTitle = QString::fromLatin1("");
+    if (getGuiDocument()->isModified()) {
+        windowTitle = QString::fromLatin1("* ");

```

Comment:
QT windows (and by extension MainWindow) have "setWindowModified"  flag that should do that automatically - can you try that?

Reply:
I've tried that but does not work. Also I can add more options in the future now with this solution.